Managers/UInputMan.cpp

```diff
@@ -228,176 +228,179 @@ int UInputMan::InputScheme::Create()
+			m_aInputMapping[INPUT_L_LEFT].SetKey(KEY_LEFT);
+			m_aInputMapping[INPUT_L_RIGHT].SetKey(KEY_RIGHT);
+			m_aInputMapping[INPUT_FIRE].SetKey(KEY_1_PAD);
+			m_aInputMapping[INPUT_AIM].SetKey(KEY_2_PAD);
+			m_aInputMapping[INPUT_AIM_UP].SetKey(KEY_UP);
+			m_aInputMapping[INPUT_AIM_DOWN].SetKey(KEY_DOWN);
+			m_aInputMapping[INPUT_PIEMENU].SetKey(KEY_3_PAD);
+			m_aInputMapping[INPUT_JUMP].SetKey(KEY_ENTER_PAD);
+			m_aInputMapping[INPUT_CROUCH].SetKey(KEY_DEL_PAD);
+			m_aInputMapping[INPUT_NEXT].SetKey(KEY_5_PAD);
+			m_aInputMapping[INPUT_PREV].SetKey(KEY_4_PAD);
+			m_aInputMapping[INPUT_WEAPON_RELOAD].SetKey(KEY_0_PAD);
+			m_aInputMapping[INPUT_WEAPON_DROP].SetKey(KEY_6_PAD);
+			m_aInputMapping[INPUT_WEAPON_PICKUP].SetKey(KEY_9_PAD);
+			m_aInputMapping[INPUT_WEAPON_CHANGE_PREV].SetKey(KEY_7_PAD);
+			m_aInputMapping[INPUT_WEAPON_CHANGE_NEXT].SetKey(KEY_8_PAD);
+			break;
 
-    }
-    // Some generic defaults; no real preset is set
-    else
-    {
-        m_SchemePreset = PRESET_NONE;
-
-        // Set up the default mouse button bindings
-        m_aInputMapping[INPUT_FIRE].SetMouseButton(MOUSE_LEFT);
-        m_aInputMapping[INPUT_PIEMENU].SetMouseButton(MOUSE_RIGHT);
-
-        // Set up the default joystick button bindings
-        m_aInputMapping[INPUT_FIRE].SetJoyButton(JOY_1);
-        m_aInputMapping[INPUT_AIM].SetJoyButton(JOY_2);
-        m_aInputMapping[INPUT_PIEMENU].SetJoyButton(JOY_3);
-        m_aInputMapping[INPUT_JUMP].SetJoyButton(JOY_4);
-        m_aInputMapping[INPUT_NEXT].SetJoyButton(JOY_6);
-        m_aInputMapping[INPUT_PREV].SetJoyButton(JOY_5);
-        m_aInputMapping[INPUT_START].SetJoyButton(JOY_8);
-        m_aInputMapping[INPUT_BACK].SetJoyButton(JOY_7);
-
-        // Set up the default joystick direction bindings
-        m_aInputMapping[INPUT_L_UP].SetDirection(0, 1, JOYDIR_ONE);
-        m_aInputMapping[INPUT_L_DOWN].SetDirection(0, 1, JOYDIR_TWO);
-        m_aInputMapping[INPUT_L_LEFT].SetDirection(0, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_L_RIGHT].SetDirection(0, 0, JOYDIR_TWO);
-        m_aInputMapping[INPUT_R_UP].SetDirection(2, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_R_DOWN].SetDirection(2, 0, JOYDIR_TWO);
-        m_aInputMapping[INPUT_R_LEFT].SetDirection(1, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_R_RIGHT].SetDirection(1, 0, JOYDIR_TWO);
-        m_aInputMapping[INPUT_FIRE].SetDirection(0, 2, JOYDIR_ONE);
-        m_aInputMapping[INPUT_JUMP].SetDirection(0, 1, JOYDIR_ONE);
-        m_aInputMapping[INPUT_CROUCH].SetDirection(0, 1, JOYDIR_TWO);
-        m_aInputMapping[INPUT_PIEMENU].SetDirection(0, 2, JOYDIR_TWO);
-
-
-        m_aInputMapping[INPUT_WEAPON_CHANGE_PREV].SetDirection(3, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_WEAPON_CHANGE_NEXT].SetDirection(3, 0, JOYDIR_TWO);
-		m_aInputMapping[INPUT_WEAPON_RELOAD].SetDirection(3, 1, JOYDIR_ONE);
-        m_aInputMapping[INPUT_WEAPON_PICKUP].SetDirection(3, 1, JOYDIR_TWO);
-    }
+		case PRESET_P3DEFAULT:
```

Comment:
I don't understand why P3 and P4 have separate defaults to P1 and P2 here, but still overlap on some things. There's no way all four players can fit on 1 keyboard (plus mouse) with this configuration, so why not just have P2, P3 and P4 share the same default keyboard configuration?
This might be better to discuss in discord instead of comments here.

Reply:
after discussion in discord, will be change in future.